Document new annotation/macro features (#636)
crystal-lang/crystal-book: docs/syntax_and_semantics/macros/README.md

define_method :foo, 1
```

### parse_type

> While most AST nodes will be obtained via either manually passed arguments, hard coded values, or retrieved from either the [type](#type-information) or [method](#method-information) information helper variables. There may be some cases where the node you need is not directly accessible, such as if you use information from different contexts to construct the path to the desired type/constant.

Reviewer: Grammar fix and minor rewording.

```suggestion
Most AST nodes are obtained via either manually passed arguments, hard coded values, or retrieved from either the [type](#type-information) or [method](#method-information) information helper variables. Yet there might be cases in which a node is not directly accessible, such as if you use information from different contexts to construct the path to the desired type/constant.
```
